### DISCUSSION TITLE
Difficult to install with docker - On plesk

### DISCUSSION THREAD

**Discussion by jxs714** [chatwoot/chatwoot#8823] [Self hosted]
Hey guys,

I followed the instructions to the T and still cannot get chatwoot to work for docker. I have redis and postgresql on my plesk server outside of docker and modified the docker-compose.yaml files to only install sidekiq and rails. Even though it connects to postgresql and redis, here are the errors Im getting for each container:

Chatwoot

> Switch to inspect mode.

Rails

> from /gems/ruby/3.2.0/gems/pg-1.5.3/lib/pg/connection.rb:824:in `connect_to_hosts'
	from /gems/ruby/3.2.0/gems/pg-1.5.3/lib/pg/connection.rb:759:in `new'
	from /gems/ruby/3.2.0/gems/pg-1.5.3/lib/pg.rb:63:in `connect'
	from /gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/postgresql_adapter.rb:78:in `new_client'
	from /gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/postgresql_adapter.rb:37:in `postgresql_connection'
	from /gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/connection_pool.rb:656:in `public_send'
	from /gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/connection_pool.rb:656:in `new_connection'
	from /gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/connection_pool.rb:700:in `checkout_new_connection'
	from /gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/connection_pool.rb:679:in `try_to_checkout_new_connection'
	from /gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/connection_pool.rb:640:in `acquire_connection'
	from /gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/connection_pool.rb:341:in `checkout'
	from /gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/connection_pool.rb:181:in `connection'
	from /gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/connection_handler.rb:211:in `retrieve_connection'
	from /gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_handling.rb:313:in `retrieve_connection'
	from /gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_handling.rb:280:in `connection'
	from /app/enterprise/app/services/features/response_bot_service.rb:20:in `vector_extension_enabled?'
	from /app/enterprise/app/models/enterprise/concerns/inbox.rb:12:in `block in <module:Inbox>'
	from /gems/ruby/3.2.0/gems/activesupport-7.0.8/lib/active_support/concern.rb:136:in `class_eval'
	from /gems/ruby/3.2.0/gems/activesupport-7.0.8/lib/active_support/concern.rb:136:in `append_features'
	from /app/config/initializers/01_inject_enterprise_edition_module.rb:78:in `include'
	from /app/config/initializers/01_inject_enterprise_edition_module.rb:78:in `block in each_extension_for'
	from /app/config/initializers/01_inject_enterprise_edition_module.rb:71:in `each'
	from /app/config/initializers/01_inject_enterprise_edition_module.rb:71:in `each_extension_for'
	from /app/config/initializers/01_inject_enterprise_edition_module.rb:42:in `include_mod_with'
	from /app/app/models/inbox.rb:192:in `<main>'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from /gems/ruby/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
	from /gems/ruby/3.2.0/gems/zeitwerk-2.6.12/lib/zeitwerk/kernel.rb:30:in `require'
	from /app/app/models/concerns/cache_keys.rb:9:in `block in <module:CacheKeys>'
	from /gems/ruby/3.2.0/gems/activesupport-7.0.8/lib/active_support/concern.rb:136:in `class_eval'
	from /gems/ruby/3.2.0/gems/activesupport-7.0.8/lib/active_support/concern.rb:136:in `append_features'
	from /app/app/models/account.rb:28:in `include'
	from /app/app/models/account.rb:28:in `<class:Account>'
	from /app/app/models/account.rb:23:in `<main>'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from /gems/ruby/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
	from /gems/ruby/3.2.0/gems/zeitwerk-2.6.12/lib/zeitwerk/kernel.rb:30:in `require'
	from /gems/ruby/3.2.0/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/helpers.rb:135:in `const_get'
	from /gems/ruby/3.2.0/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/helpers.rb:135:in `cget'
	from /gems/ruby/3.2.0/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/eager_load.rb:182:in `block in actual_eager_load_dir'
	from /gems/ruby/3.2.0/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/helpers.rb:40:in `block in ls'
	from /gems/ruby/3.2.0/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/helpers.rb:25:in `each'
	from /gems/ruby/3.2.0/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/helpers.rb:25:in `ls'
	from /gems/ruby/3.2.0/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/eager_load.rb:170:in `actual_eager_load_dir'
	from /gems/ruby/3.2.0/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/eager_load.rb:17:in `block (2 levels) in eager_load'
	from /gems/ruby/3.2.0/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/eager_load.rb:16:in `each'
	from /gems/ruby/3.2.0/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/eager_load.rb:16:in `block in eager_load'
	from /gems/ruby/3.2.0/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/eager_load.rb:10:in `synchronize'
	from /gems/ruby/3.2.0/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/eager_load.rb:10:in `eager_load'
	from /gems/ruby/3.2.0/gems/zeitwerk-2.6.12/lib/zeitwerk/loader.rb:379:in `block in eager_load_all'
	from /gems/ruby/3.2.0/gems/zeitwerk-2.6.12/lib/zeitwerk/loader.rb:377:in `each'
	from /gems/ruby/3.2.0/gems/zeitwerk-2.6.12/lib/zeitwerk/loader.rb:377:in `eager_load_all'
	from /gems/ruby/3.2.0/gems/railties-7.0.8/lib/rails/application/finisher.rb:74:in `block in <module:Finisher>'
	from /gems/ruby/3.2.0/gems/railties-7.0.8/lib/rails/initializable.rb:32:in `instance_exec'
	from /gems/ruby/3.2.0/gems/railties-7.0.8/lib/rails/initializable.rb:32:in `run'
	from /gems/ruby/3.2.0/gems/railties-7.0.8/lib/rails/initializable.rb:61:in `block in run_initializers'
	from /usr/local/lib/ruby/3.2.0/tsort.rb:228:in `block in tsort_each'
	from /usr/local/lib/ruby/3.2.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
	from /usr/local/lib/ruby/3.2.0/tsort.rb:431:in `each_strongly_connected_component_from'
	from /usr/local/lib/ruby/3.2.0/tsort.rb:349:in `block in each_strongly_connected_component'
	from /usr/local/lib/ruby/3.2.0/tsort.rb:347:in `each'
	from /usr/local/lib/ruby/3.2.0/tsort.rb:347:in `call'
	from /usr/local/lib/ruby/3.2.0/tsort.rb:347:in `each_strongly_connected_component'
	from /usr/local/lib/ruby/3.2.0/tsort.rb:226:in `tsort_each'
	from /usr/local/lib/ruby/3.2.0/tsort.rb:205:in `tsort_each'
	from /gems/ruby/3.2.0/gems/railties-7.0.8/lib/rails/initializable.rb:60:in `run_initializers'
	from /gems/ruby/3.2.0/gems/railties-7.0.8/lib/rails/application.rb:372:in `initialize!'
	from /app/config/environment.rb:5:in `<main>'
	from config.ru:3:in `require_relative'
	from config.ru:3:in `block in <main>'
	from /gems/ruby/3.2.0/gems/rack-2.2.8/lib/rack/builder.rb:116:in `eval'
	from /gems/ruby/3.2.0/gems/rack-2.2.8/lib/rack/builder.rb:116:in `new_from_string'
	from /gems/ruby/3.2.0/gems/rack-2.2.8/lib/rack/builder.rb:105:in `load_file'
	from /gems/ruby/3.2.0/gems/rack-2.2.8/lib/rack/builder.rb:66:in `parse_file'
	from /gems/ruby/3.2.0/gems/rack-2.2.8/lib/rack/server.rb:349:in `build_app_and_options_from_config'
	from /gems/ruby/3.2.0/gems/rack-2.2.8/lib/rack/server.rb:249:in `app'
	from /gems/ruby/3.2.0/gems/rack-2.2.8/lib/rack/server.rb:422:in `wrapped_app'
	from /gems/ruby/3.2.0/gems/rack-2.2.8/lib/rack/server.rb:312:in `block in start'
	from /gems/ruby/3.2.0/gems/rack-2.2.8/lib/rack/server.rb:379:in `handle_profiling'
	from /gems/ruby/3.2.0/gems/rack-2.2.8/lib/rack/server.rb:311:in `start'
	from /gems/ruby/3.2.0/gems/railties-7.0.8/lib/rails/commands/server/server_command.rb:38:in `start'
	from /gems/ruby/3.2.0/gems/railties-7.0.8/lib/rails/commands/server/server_command.rb:143:in `block in perform'
	from <internal:kernel>:90:in `tap'
	from /gems/ruby/3.2.0/gems/railties-7.0.8/lib/rails/commands/server/server_command.rb:134:in `perform'
	from /gems/ruby/3.2.0/gems/thor-1.3.0/lib/thor/command.rb:28:in `run'
	from /gems/ruby/3.2.0/gems/thor-1.3.0/lib/thor/invocation.rb:127:in `invoke_command'
	from /gems/ruby/3.2.0/gems/thor-1.3.0/lib/thor.rb:527:in `dispatch'
	from /gems/ruby/3.2.0/gems/railties-7.0.8/lib/rails/command/base.rb:87:in `perform'
	from /gems/ruby/3.2.0/gems/railties-7.0.8/lib/rails/command.rb:48:in `invoke'
	from /gems/ruby/3.2.0/gems/railties-7.0.8/lib/rails/commands.rb:18:in `<main>'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from /gems/ruby/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
	from bin/rails:4:in `<main>'

Sidekiq

> {"ts":"2024-01-31T03:59:16.612Z","pid":1,"tid":"5bp","lvl":"INFO","msg":"Sidekiq 7.2.0 connecting to Redis with options {:size=>10, :pool_name=>\"internal\", :url=>\"redis://172.17.0.1:6379\", :password=>\"REDACTED\", :reconnect_attempts=>2, :timeout=>1}"}
{"ts":"2024-01-31T03:59:16.631Z","pid":1,"tid":"5bp","lvl":"INFO","msg":"Cron Jobs - added job with name: internal_check_new_versions_job"}
{"ts":"2024-01-31T03:59:16.636Z","pid":1,"tid":"5bp","lvl":"INFO","msg":"Cron Jobs - added job with name: trigger_scheduled_items_job"}
{"ts":"2024-01-31T03:59:16.640Z","pid":1,"tid":"5bp","lvl":"INFO","msg":"Cron Jobs - added job with name: trigger_imap_email_inboxes_job"}
{"ts":"2024-01-31T03:59:16.643Z","pid":1,"tid":"5bp","lvl":"INFO","msg":"Cron Jobs - added job with name: remove_stale_contact_inboxes_job.rb"}
{"ts":"2024-01-31T03:59:20.977Z","pid":1,"tid":"5bp","lvl":"INFO","msg":"Booted Rails 7.0.8 application in production environment"}
{"ts":"2024-01-31T03:59:20.977Z","pid":1,"tid":"5bp","lvl":"INFO","msg":"Running in ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x86_64-linux-musl]"}
{"ts":"2024-01-31T03:59:20.977Z","pid":1,"tid":"5bp","lvl":"INFO","msg":"See LICENSE and the LGPL-3.0 for licensing details."}
{"ts":"2024-01-31T03:59:20.977Z","pid":1,"tid":"5bp","lvl":"INFO","msg":"Upgrade to Sidekiq Pro for more features and support: https://sidekiq.org"}
{"ts":"2024-01-31T04:00:02.548Z","pid":1,"tid":"5bp","lvl":"INFO","msg":"Sidekiq 7.2.0 connecting to Redis with options {:size=>10, :pool_name=>\"internal\", :url=>\"redis://172.17.0.1:6379\", :password=>\"REDACTED\", :reconnect_attempts=>2, :timeout=>1}"}
{"ts":"2024-01-31T04:00:02.553Z","pid":1,"tid":"5bp","lvl":"INFO","msg":"Cron Jobs - added job with name: internal_check_new_versions_job"}
{"ts":"2024-01-31T04:00:02.556Z","pid":1,"tid":"5bp","lvl":"INFO","msg":"Cron Jobs - added job with name: trigger_scheduled_items_job"}
{"ts":"2024-01-31T04:00:02.559Z","pid":1,"tid":"5bp","lvl":"INFO","msg":"Cron Jobs - added job with name: trigger_imap_email_inboxes_job"}
{"ts":"2024-01-31T04:00:02.562Z","pid":1,"tid":"5bp","lvl":"INFO","msg":"Cron Jobs - added job with name: remove_stale_contact_inboxes_job.rb"}
{"ts":"2024-01-31T04:00:04.713Z","pid":1,"tid":"5bp","lvl":"INFO","msg":"Booted Rails 7.0.8 application in production environment"}
{"ts":"2024-01-31T04:00:04.713Z","pid":1,"tid":"5bp","lvl":"INFO","msg":"Running in ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x86_64-linux-musl]"}
{"ts":"2024-01-31T04:00:04.713Z","pid":1,"tid":"5bp","lvl":"INFO","msg":"See LICENSE and the LGPL-3.0 for licensing details."}
{"ts":"2024-01-31T04:00:04.713Z","pid":1,"tid":"5bp","lvl":"INFO","msg":"Upgrade to Sidekiq Pro for more features and support: https://sidekiq.org"}
/gems/ruby/3.2.0/gems/gmail_xoauth-0.4.2/lib/gmail_xoauth/imap_xoauth_authenticator.rb:31: warning: Net::IMAP.add_authenticator is deprecated.  Use Net::IMAP::SASL.add_authenticator instead.
/gems/ruby/3.2.0/gems/gmail_xoauth-0.4.2/lib/gmail_xoauth/imap_xoauth2_authenticator.rb:24: warning: Net::IMAP.add_authenticator is deprecated.  Use Net::IMAP::SASL.add_authenticator instead.
fatal: not a git repository (or any of the parent directories): .git
You are connecting to Redis 6.0.16, Sidekiq requires Redis 6.2.0 or greater
/gems/ruby/3.2.0/gems/sidekiq-7.2.0/lib/sidekiq/cli.rb:77:in `run'
/gems/ruby/3.2.0/gems/sidekiq-7.2.0/bin/sidekiq:31:in `<top (required)>'
/gems/ruby/3.2.0/bin/sidekiq:25:in `load'
/gems/ruby/3.2.0/bin/sidekiq:25:in `<top (required)>'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/cli/exec.rb:58:in `load'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/cli/exec.rb:58:in `kernel_load'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/cli/exec.rb:23:in `run'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/cli.rb:451:in `exec'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/vendor/thor/lib/thor/command.rb:28:in `run'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/vendor/thor/lib/thor.rb:527:in `dispatch'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/cli.rb:34:in `dispatch'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/vendor/thor/lib/thor/base.rb:584:in `start'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/cli.rb:28:in `start'
/usr/local/bundle/gems/bundler-2.5.4/exe/bundle:28:in `block in <top (required)>'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
/usr/local/bundle/gems/bundler-2.5.4/exe/bundle:20:in `<top (required)>'
/usr/local/bundle/bin/bundle:25:in `load'
/usr/local/bundle/bin/bundle:25:in `<main>'
/gems/ruby/3.2.0/gems/gmail_xoauth-0.4.2/lib/gmail_xoauth/imap_xoauth_authenticator.rb:31: warning: Net::IMAP.add_authenticator is deprecated.  Use Net::IMAP::SASL.add_authenticator instead.
/gems/ruby/3.2.0/gems/gmail_xoauth-0.4.2/lib/gmail_xoauth/imap_xoauth2_authenticator.rb:24: warning: Net::IMAP.add_authenticator is deprecated.  Use Net::IMAP::SASL.add_authenticator instead.
fatal: not a git repository (or any of the parent directories): .git
You are connecting to Redis 6.0.16, Sidekiq requires Redis 6.2.0 or greater
/gems/ruby/3.2.0/gems/sidekiq-7.2.0/lib/sidekiq/cli.rb:77:in `run'
/gems/ruby/3.2.0/gems/sidekiq-7.2.0/bin/sidekiq:31:in `<top (required)>'
/gems/ruby/3.2.0/bin/sidekiq:25:in `load'
/gems/ruby/3.2.0/bin/sidekiq:25:in `<top (required)>'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/cli/exec.rb:58:in `load'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/cli/exec.rb:58:in `kernel_load'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/cli/exec.rb:23:in `run'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/cli.rb:451:in `exec'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/vendor/thor/lib/thor/command.rb:28:in `run'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/vendor/thor/lib/thor.rb:527:in `dispatch'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/cli.rb:34:in `dispatch'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/vendor/thor/lib/thor/base.rb:584:in `start'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/cli.rb:28:in `start'
/usr/local/bundle/gems/bundler-2.5.4/exe/bundle:28:in `block in <top (required)>'
/usr/local/bundle/gems/bundler-2.5.4/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
/usr/local/bundle/gems/bundler-2.5.4/exe/bundle:20:in `<top (required)>'
/usr/local/bundle/bin/bundle:25:in `load'
/usr/local/bundle/bin/bundle:25:in `<main>'

Been trying this for 8+ hours and cant get it to work. Any insight?

**Comment by jxs714** [comment]
I guess theres really no support here with solutions. I fixed the issue myself. You will need to customize the docker-compose.yaml file and remove redis and postgresql. Youll need to enter the correct local docker ip as the hostname (for me it was 172.17.0.1 for both redis and postgresql) When you install the docker and all containers as the tutorial says, you only need the RAILS one that it generates. I got the software to work, but now my issue is emails are not working for "recover password" for users. No matter what settings I enter in the .env for SSL / TLS , address etc, it doesnt work. When it did go through and confirm on the GUI that the email was sent, nothing comes into my inbox or goes through postfix. Tried all kinds of combinations, but I see that the emails stay stuck in Sidekiq, which leaves it in the enqueued area. Not sure what to do and it seems others are having this issue and labeling it a bug.

**Comment by sojan-official** [comment]
@amaljosea  please follow the instructions to debug issues with email : https://www.chatwoot.com/docs/self-hosted/faq#how-to-debug-smtp-email-errors-in-chatwoot

We recommend using a provider like sendgrid, mailgun or ses 
https://www.chatwoot.com/docs/self-hosted/configuration/environment-variables#configure-emails

**Comment by jxs714** [comment]
Hello

Thanks for the response. I've did the .env configuration and it still doesn't work. I also tried postmark with api keys. It doesn't work either. It seemse Sidekiq is just not responsive somehow.
